File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;
import java.io.*;

class Solution {
    public String solution(String[] participant, String[] completion) {
        String answer = "";
        // 1. Hash map을 만든다.
        HashMap<String, Integer> map = new HashMap<>();
        for(String player: participant){
            map.put(player, map.getOrDefault(player,0) + 1);
        }
        // 2. Hash map을 뺀다.
        for(String player: completion)
            map.put(player,map.get(player) - 1);
        // 3. Value가 0이 아닌 마지막 주자를 찾는다.
        for(String key : map.keySet()){
            if(map.get(key) != 0){
                answer = key;
                break;
            }
        }
        return answer;
    }
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participant, completion):
    answer = ""

    participant.sort()
    completion.sort()
    
    # print(participant)
    # print(completion)
    
    for i in range(len(completion)):
        if participant[i] != completion[i]:
            answer = participant[i]
            return answer
            
    answer = participant.pop()
    
    return answer